### DISCUSSION TITLE
Can you get Parameterized telemetry?

### DISCUSSION THREAD

**Discussion by Overloader6** [nasa/openmct#4176] [Community Q&A] [answered]
So this might sound like a fairly odd question but the API i'm using to get telemetry allows me to request specific telemetry by doing for example `r.resource["Example Resource"]` Do I have to hardcode that parameter or can I somehow let the user choose the parameter when he drags the telemetry into their Layout?

**Comment by akhenry** [comment, marked as answer]
The object tree is resolved at client load time by either querying a static dictionary file, or by sending a request to a server for available telemetry points. In either case, the general idea is that Open MCT knows at runtime what telemetry points are available, and presents them in a tree. The user finds the telemetry point they are interested in by either navigating the tree's hierarchy, or by using the search feature, and then drags the telemetry into their view.

So, out of the box, Open MCT assumes you have some way of knowing up front what telemetry is available. We don't have any UI that does what you describe. **However**, there is an undocumented feature that, while it's not it's intended purpose, might give you want you want.

If you add the following to your telemetry metadata...
```
filters: ['equals'],
```
and then drag that telemetry point into a plot or table, when you edit the plot or table you will see a new "DATA FILTERS" section in the inspector on the right. This will allow you to enter some text that will be passed to your telemetry provider as an option with any requests or subscriptions for that telemetry. eg.

<img src="https://user-images.githubusercontent.com/4215777/131760469-b9c29293-3b00-42c4-b273-d8b7729ecca2.png" width="256">

Here's an example of how you [enable a filter field for the `cos` field of our SineWaveGenerator example telemetry](https://github.com/nasa/openmct/commit/248f160e738ba411aeb02836318063f21d5a3ca9#diff-d03bdadde4d61e2fed747d149e17ccdb8754af38fcbd084318d5d17ce66d66d5R48).

If you then set a breakpoint in [the sinewave generator's telemetry provider](https://github.com/nasa/openmct/blob/master/example/generator/GeneratorProvider.js), you can see the filter value being passed all the way through:

![Screen Shot 2021-09-01 at 4 50 43 PM](https://user-images.githubusercontent.com/4215777/131759987-60257e4c-f70d-4dd1-a0f4-ff9cf14e756c.png)

You could use this filter value to build a request to your telemetry backend for a user-specified telemetry point.

A few caveats, this approach will only work with plots and tables, and will be persisted with those objects, and not with the telemetry point itself.
